src/eth/storage/postgres/postgres.rs

```diff
@@ -308,8 +311,53 @@ impl EthStorage for Postgres {
         todo!()
     }
 
-    async fn read_logs(&self, _: &LogFilter) -> anyhow::Result<Vec<LogMined>> {
-        Ok(Vec::new())
+    async fn read_logs(&self, filter: &LogFilter) -> anyhow::Result<Vec<LogMined>> {
+        let from: i64 = filter.from_block.try_into().unwrap();
+        let to: i64;
+        let query = r#"
+            SELECT
+                address as "address:  "
+                , data as "data:  "
+                , transaction_hash as "transaction_hash: "
+                , log_idx as "log_idx:  "
+                , block_number as "block_number:  "
+                , block_hash as "block_hash:  "
+            FROM logs
+            WHERE block_number >= $1
+        "#
+        .to_owned();
+        let builder = &mut QueryBuilder::new(query);
+        builder.push_bind(from);
+
+        // verifies if to_block exists
+        if let Some(block_number) = filter.to_block {
+            builder.push(" AND block_number <= $2");
+            to = block_number.try_into().unwrap();
```

Comment:
use `?` instead of `unwrap()`
```suggestion
            to = block_number.try_into()?;
```

Reply:
Done